torch/distributed/checkpoint/_fsspec_filesystem.py

```diff
@@ -257,8 +266,12 @@ def _write_files_from_queue(
     result_queue: queue.Queue,
     planner: SavePlanner,
     inflight_threshhold: int,
-    fs: AbstractFileSystem,
+    fs: "AbstractFileSystem",
 ) -> None:
+    try:
```

Comment:
Nit: Ideally we could define a private `_fsspec_open` with the try catch included, which could remove some of the duplication from this file.